marshmallow/schema.py

```diff
@@ -525,6 +525,16 @@ def dump(self, obj, many=None, update_fields=True, **kwargs):
             error_handler = self.handle_error or self.__error_handler__
             if callable(error_handler):
                 error_handler(errors, obj)
+            exc = ValidationError(
+                errors,
+                field_names=self._marshal.error_field_names,
+                fields=self._marshal.error_fields,
+                data=result,
+                **self._marshal.error_kwargs
+            )
+            self.handle_error(exc, result)
```

Comment:
> without even checking if self.handle_error is callable.

It is callable, I checked it https://github.com/marshmallow-code/marshmallow/blob/dev/marshmallow/schema.py#L384
If it is overridden with non-callable, it is an error and "Errors should never pass silently." 🐍 

> now you call self.__error_handler__, and then self.handle_error

I do not see why anybody would have both of those. Keeping calls separate helps to rip out deprecated `__error_handler__ ` when it is time.